src/openms_gui/source/VISUAL/Spectrum2DCanvas.cpp

```diff
@@ -1072,11 +1072,9 @@ namespace OpenMS
     {
       gradient_str = linear_gradient_.toString();
     }
-    for (Size i = 0; i < layers_.size(); ++i)
-    {
-      layers_[i].param.setValue("dot:gradient", gradient_str);
-      recalculateDotGradient_(i);
-    }
+    layers_[layers_.size()-1].param.setValue("dot:gradient", gradient_str);
+    recalculateDotGradient_(layers_.size()-1);
```

Comment:
@lars20070 the `recalculateDotGradient_` function updates the gradients to a common maximal and minimal value and ensures that the same color for the same gradient has the same interpretation.
You could try to put this one line back in the loop. This will update the gradient ranges and might change the colors a bit. But it should keep the selected color gradients for each layer as long as you don't reset the `dot:gradient` param.